logic/resolvable/Concludable.java

```diff
@@ -188,8 +188,13 @@ boolean unificationSatisfiable(ThingVariable concludableThingVar, ThingVariable
         }
 
         if (!concludableThingVar.value().isEmpty() && !conclusionThingVar.value().isEmpty()) {
-            // TODO: detect value contradictions between constant predicates
-            satisfiable &= true;
+            assert conclusionThingVar.value().size() == 1;
+            ValueConstraint<?> conclusionValueConstraint = conclusionThingVar.value().stream().findFirst().get();
+
+            assert conclusionValueConstraint.predicate().isEquality() &&
+                    conclusionValueConstraint.predicate().asEquality().equals(EQ);
+
+            satisfiable &=  concludableThingVar.value().stream().allMatch( v -> v.maybeCompatible(conclusionValueConstraint) );
```

Comment:
Yes. A lot of them can be static.

A lot of files have spacing which gets fixed by the auto-formatter and would lead to random lines being updated.
I do try to run it on my changes.